distributed/diagnostics/tests/test_scheduler_plugin.py

```diff
@@ -117,7 +117,8 @@ async def start(self, scheduler):
     s.add_plugin(plugin)
     s.add_plugin(plugin, name="another")
     with pytest.raises(ValueError) as excinfo:
-        s.remove_plugin(plugin)
+        with pytest.warns(UserWarning, match="Removing scheduler plugins by value"):
```

Comment:
Hmm based on 

https://github.com/dask/distributed/blob/dc471862b7efc8473583e7e96b6fcaf8d5c7c527/distributed/scheduler.py#L5507-L5511

I would have expected this to emit a `FutureWarning` instead of a `UserWarning`

Reply:
Agreed- I just made the adjustment. I wonder why it passed before

Reply:
Interesting. This might be a bug in `pytest` (I opened up https://github.com/pytest-dev/pytest/issues/9036). If you change the order of `pytest.raises` and `pytest.warns`, then things behave as expected